### PR TITLE
Do not access HIR to check impl wf.

### PR DESCRIPTION
**Pull request by cjgillot**
r? @ghost

### CODE REVIEW

File: compiler/rustc_typeck/src/impl_wf_check.rs [hunk 1]
```diff
@@ -216,33 +208,32 @@ fn report_unused_parameter(tcx: TyCtxt<'_>, span: Span, kind: &str, name: &str)
...
         };
-        match seen_items.entry(impl_item.ident.normalize_to_macros_2_0()) {
+        let span = tcx.def_span(impl_item_ref);
+        let ident = impl_item.ident(tcx);
+        match seen_items.entry(ident.normalize_to_macros_2_0()) {
             Occupied(entry) => {
                 let mut err = struct_span_err!(
                     tcx.sess,
-                    impl_item.span,
+                    tcx.def_span(impl_item_ref),
```
[the remark this answers is not in the document]
Comment (deltragon) [reply compiler/rustc_typeck/src/impl_wf_check.rs:226]: Probably still needs rustfmt after this, might fit on a single line now

**Comment by cjgillot** [comment]
r? rust-lang/compiler

**Comment by petrochenkov** [comment]
r=me with https://github.com/rust-lang/rust/pull/98396#discussion_r904159229 addressed.
@rustbot author

**Pull request merged**